src/components/select-menu/useSelectMenuAnimations.tsx

```diff
@@ -0,0 +1,199 @@
+import * as React from 'react';
+import useReducedMotion from '../utils/useReducedMotion';
+
+type UseSelectMenuAnimationsPropsType = {
+  selectId: string;
+  popupClassName: string;
+  popupContentClassName: string;
+  floatingContainerClassName: string;
+  selectElementClassName: string;
+};
+
+const MIN_POPUP_WIDTH = 120;
+const ANIMATE_CLASSNAME = 'animate-on-transforms';
```

Comment:
hmm not sure if we shouldn't use `sg-*` prefix here as in ohter global classes

Reply:
It was used in the figma file, but I can totally change it to eng